src/guard.ts

```diff
@@ -5,16 +5,21 @@ import { AUTH0_TOKEN } from './token';
 import type { Auth0VueClient } from './interfaces';
 import type { App } from 'vue';
 import { unref } from 'vue';
+import { RedirectLoginOptions } from '@auth0/auth0-spa-js';
 
-async function createGuardHandler(client: Auth0VueClient, to: RouteLocation) {
+async function createGuardHandler(client: Auth0VueClient, to: RouteLocation, redirectLoginOptions?: RedirectLoginOptions) {
+  const defaultRedirectLoginOptions = { appState: { target: to.fullPath } };
   const fn = async () => {
     if (unref(client.isAuthenticated)) {
       return true;
     }
 
-    await client.loginWithRedirect({
-      appState: { target: to.fullPath }
-    });
+    if (!redirectLoginOptions) {
+        redirectLoginOptions = defaultRedirectLoginOptions;
+    } else if (redirectLoginOptions && !redirectLoginOptions.appState) {
+        redirectLoginOptions.appState = defaultRedirectLoginOptions.appState;
+    }
+    await client.loginWithRedirect(redirectLoginOptions);
```

Comment:
Let's keep this as it was, and spread the options

```suggestion
    await client.loginWithRedirect({
      appState: { target: to.fullPath },
      ...redirectLoginOptions
    });
```